README_python.md

```diff
@@ -170,35 +200,240 @@ essentially zero cost, with storage being shared. Thus, we can usually pass it
+the existing Python API, while an `hl.InputBuffer` is a factory for
+`hl.ImageParam`.
+
+From the Generator author's perspective, a field initialized with `InputScalar`
+**is** a `Param` – not kinda-like-one, not a magic wrapper that forwards
+everything; it is literally just `hl.Param`. Similarly, an `InputBuffer`
+produces `ImageParam`, and an `InputFunc` is a wrapper around `Func`. You won't
+be able to assign a new value to the member field for Inputs – as with
+GeneratorParams, they are "read-only" to the Generator – but you will be able to
+set constraints on them.
+
+Note that in addition to specifying a concrete type and dimensionality for the
+inputs, these factory classes support the ability to specify either (or both)
+`None`, which means the type/dimensionality will be provided by GeneratorParams
+in the build system.
+
+#### hl.OutputBuffer, hl.OutputScalar
+
+These declare the output(s) of the Pipeline that the Generator will produce. An
+`hl.OutputBuffer` is, essentially, a "factory" that produces an `hl.Func` in the
+existing Python API. (`hl.OutputScalar` is just an `hl.OutputBuffer` that always
+has zero dimensions.)
+
+From the Generator author's perspective, a field declared with `OutputBuffer`
+**is** a `Func` – not kinda-like-one, not a magic wrapper that forwards
+everything; it is literally just `hl.Func` (with type-and-dimensionality set to
+match, see recent PR https://github.com/halide/Halide/pull/6734) . You won't be
+able to assign a new value to the member field for Inputs – as with
+GeneratorParams, they are "read-only" to the Generator – but you will be able to
+set constraints on them.
+
+Note that in addition to specifying a concrete type and dimensionality for the
+inputs, these factory classes support the ability to specify either (or both) as
+`None`, which means the type/dimensionality will be provided by GeneratorParams
+in the build system.
+
+#### Names
+
+Note that all of the GeneratorParams, Inputs, and Outputs have names that are
+implicitly filled in based on the fieldname of their initial assignment; unlike
+in C++ Generators, there isn't a way to "override" this name (i.e., the name in
+the IR will always exactly match the Python field name). Names have the same
+constraints as for C++ Generators (essentially, a C identifier, but without an
+initial underscore, and without any double underscore anywhere).
+
+#### generate() method
+
+This will be called by the Generator machinery to build the Pipeline. As with
+C++ Generators, the only required task is to ensure that all Output fields are
+fully defined, in a way that matches the type-and-dimension constraints
+specified.
+
+It is required that the `generate()` method be defined by the Generator.
+
+(Note that, by convention, Halide Generators use `g` instead of `self` in their
+`generate()` method to make the expression language terser; this is not in any
+way required, but is recommended.)
```

Comment:
Whoops, I missed this earlier. This needs to be called out more; somewhere higher up than four heading levels deep.

Reply:
See above